flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/utils/DynamicPartitionPruningUtils.java

```diff
@@ -234,20 +234,16 @@ private static boolean isSuitableFilter(RexNode filterCondition) {
         }
 
         private void setTables(ContextResolvedTable catalogTable) {
-            if (tables.size() == 0) {
-                tables.add(catalogTable);
-            } else {
-                boolean hasAdded = false;
-                for (ContextResolvedTable thisTable : new ArrayList<>(tables)) {
-                    if (hasAdded) {
-                        break;
-                    }
-                    if (!thisTable.getIdentifier().equals(catalogTable.getIdentifier())) {
-                        tables.add(catalogTable);
-                        hasAdded = true;
-                    }
+            boolean alreadyExists = false;
+            for (ContextResolvedTable table : tables) {
```

Comment:
Not sure how big the tables could be and how often the method will be called. Does it make sense to use e.g. a Map<K,V> to avoid the loop?

Reply:
Good point. But note that `tables` is of type `Set` already